theatre/studio/src/StudioStore/StudioStore.ts

```diff
@@ -141,46 +142,62 @@ export default class StudioStore {
                   .getValue(),
               )
 
-              forEachDeep(
-                value,
-                (v, pathToProp) => {
-                  if (typeof v === 'undefined' || v === null) {
-                    return
-                  }
-
-                  const propAddress = {...root.address, pathToProp}
-
-                  const trackId = get(
-                    sequenceTracksTree,
-                    pathToProp,
-                  ) as $FixMe as SequenceTrackId | undefined
-
-                  if (typeof trackId === 'string') {
-                    const propConfig = get(
-                      root.template.config.props,
-                      pathToProp,
-                    ) as PropTypeConfig | undefined
-                    if (propConfig?.sanitize) v = propConfig.sanitize(v)
-
-                    const seq = root.sheet.getSequence()
-                    seq.position = seq.closestGridPosition(seq.position)
-                    stateEditors.coreByProject.historic.sheetsById.sequence.setKeyframeAtPosition(
-                      {
-                        ...propAddress,
-                        trackId,
-                        position: seq.position,
-                        value: v as $FixMe,
-                        snappingFunction: seq.closestGridPosition,
-                      },
-                    )
-                  } else {
-                    stateEditors.coreByProject.historic.sheetsById.staticOverrides.byObject.setValueOfPrimitiveProp(
-                      {...propAddress, value: v},
-                    )
-                  }
-                },
-                getPointerParts(pointer as Pointer<$IntentionalAny>).path,
-              )
+              const propConfig = get(
+                root.template.config.props,
+                path,
+              ) as PropTypeConfig
+
+              const setStaticOrKeyframeProp = <T>(
+                value: T,
+                path: PathToProp,
+              ) => {
+                if (typeof value === 'undefined' || value === null) {
+                  return
+                }
+
+                const propAddress = {...root.address, pathToProp: path}
+
+                const trackId = get(sequenceTracksTree, path) as $FixMe as
+                  | SequenceTrackId
+                  | undefined
+                if (typeof trackId === 'string') {
+                  const propConfig = get(root.template.config.props, path) as
```

Comment:
I think this wouldn't work for compound props. Because compound props' config looks like this `{type: 'compound', props: {}}`. `get()` wouldn't traverse the `props` prop. Use `getPropConfigByPath()` instead: https://github.com/AriaMinaei/theatre/blob/b97f51270e3cd3ab6c3fb607ef88939c0e89a0a5/theatre/shared/src/propTypes/utils.ts#L15

Reply:
You're right, let me see

Reply:
you were naughty too, this code is from you :D